### PR TITLE
More Beta Changes

### PR DESCRIPTION
**Pull request by LucaHaverty**
### Description
Some more changes that were requested by users during beta testing day.

### Changes
- [x] Center cam when a robot is removed
- [x] Remove 'alpha' text in controls panel
- [x] 'Esc' to close the active modal
- [x] Remove .mira file extension in add object modal
- [x] Make tooltips last longer
- [x] UI gradient updates - made some gradients backwards like on Jira:
image
- [x] Theme editor updates - make it clear that the default theme cannot be edited, button locations for deleting are more intuitive, color values round to the nearest int
- [x]  Make robots not appear under the floor during robot placement
- [x] Fix weird scaling of the spawn location panel with different aspect ratios

### CODE REVIEW

**Review by a-crowell — APPROVED** [review]
It all works well, nice work!

File: engine/Assets/Scripts/UI/Dynamic/Modals/Spawning/AddRobotModal.cs [hunk 1]
```diff
@@ -40,10 +40,12 @@ public class AddRobotModal : ModalDynamic {
...
             });
 
-            var chooseRobotDropdown = MainContent.CreateDropdown()
-                                          .SetOptions(_files.Select(x => Path.GetFileName(x)).ToArray())
-                                          .AddOnValueChangedEvent((d, i, data) => _selectedIndex = i)
-                                          .SetTopStretch<Dropdown>();
+            var chooseRobotDropdown =
+                MainContent.CreateDropdown()
+                    .SetOptions(
+                        _files.Select(x => Path.GetFileName(x).Substring(0, Path.GetFileName(x).Length - 5)).ToArray())
```
Comment (BrandonPacewic) [inline engine/Assets/Scripts/UI/Dynamic/Modals/Spawning/AddRobotModal.cs:46]: I feel like we could get some unintended consequences from doing this. However all the files should be `.mira` anyways so I think it should be fine.

Comment (KyroVibe) [reply engine/Assets/Scripts/UI/Dynamic/Modals/Spawning/AddRobotModal.cs:46]: I'd use this instead: https://learn.microsoft.com/en-us/dotnet/api/system.io.path.getfilenamewithoutextension?view=net-7.0

That being said, I changed where this code is stored in my multiplayer branch it prolly won't matter.

**Review by KyroVibe — APPROVED** [review]

**Comment by KyroVibe** [comment]
Will the removal of "alpha" work for the tooltips as well?

**Review by Markytt — APPROVED** [review]

**Comment by LucaHaverty** [comment]
> Will the removal of "alpha" work for the tooltips as well?

It should, assuming it uses the same code to generate the names. That feature was added in Ayla's controls branch I believe, so we can check when that is merged.

**Review by BrandonPacewic — APPROVED** [review]
Nice and easy. Will make a note to check the keycode names in the tooltips later.

**Review by KyroVibe — CHANGES_REQUESTED** [review]
image
There isn't a way to continue to the next step in match mode

**Comment by LucaHaverty** [comment]
> Should the theme editor panel tween out and back every time you change the selected theme save?

Good point, that has been changed now

**Review by KyroVibe — CHANGES_REQUESTED** [review]
While the accept button now shows up, the accept button on the robot placement panel doesn't work.

Also could you just disable the entire MainHUD while setting up robots and scoring zones in match mode?

**Review by KyroVibe — APPROVED** [review]
Button works. Good work

**Pull request merged**